lenskit/algorithms/basic.py

```diff
@@ -153,6 +153,60 @@ def inverse_transform(self, ratings):
             rvps['rating'] += rvps['u_off'].fillna(0)
         return rvps.drop(columns=['u_off', 'i_off'])
 
+    def transform_user(self, ratings):
+        """
+        Transform a user's ratings by subtracting the bias model.
+
+        Args:
+            ratings(pandas.Series): The user's ratings, indexed by item.
+                Must have at least `item` as index and `rating` column.
+
+        Returns:
+            pandas.Series:
+                The transformed ratings and the user bias.
+        """
+        ratings = ratings.subtract(self.mean_)
+
+        # build user_offsets_
+        if self.item_offsets_ is not None:
+            rv = ratings.values - self.item_offsets_[
+                self.item_offsets_.index.isin(ratings.index)].values
```

Comment:
I don't know that this code is correct - it doesn't do anything to ensure the two arrays are in the same order.

Consider:

```python
ioff = self.item_offsets_.reindex(ratings.index, fill_value=0)
rv = ratings - ioff
```
```